### ISSUE TITLE
Amplify Console Environment Variables override to only set variable on specific branches

### ISSUE THREAD

**Issue by arledesma** [aws-amplify/amplify-console#2378]
**Please describe which feature you have a question about?**

We would like to set Cognito related [amplify console environment variables](https://docs.aws.amazon.com/amplify/latest/userguide/environment-variables.html#amplify-console-environment-variables) only as overrides, not applying these values to `All Branches`.

We would like to be able to use these environment variables in isolation without affecting other branches, especially our production branch.

How can we set set environment variables like `AMPLIFY_USERPOOL_ID` and `AMPLIFY_IDENTITYPOOL_ID` only on specific branches inside of the Amplify Environment Variables section of an Amplify App?

**Provide additional details**

This would be the inverse of #1202 where they were asking how to unset an environment variable.  If this was possible then we could unset other branches, but this would be tedious and error prone with new preview builds.

This can be accomplished in the amplify buildSpec but seems like the wrong place to add this type of logic if Amplify does have the ability to only set the variable on specific branches.

**What AWS Services are you utilizing?**

N/A

**Provide additional details e.g. code snippets**

N/A

**Comment by hloriana** [comment]
Hi @arledesma you can set branch-specific values for environment variables in the console. See step 5 under **Set environment variables**: https://docs.aws.amazon.com/amplify/latest/userguide/environment-variables.html

**Comment by arledesma** [comment]
@hloriana I'm afraid that I may have not articulated my question very well.

I only want to overwrite for a specific branch and have no value for all other branches.  These are amplify defined variables that I assume will cause different side effects depending on their value.  We do not want to have a value defined for any branches except one branch.

**Comment by hloriana** [comment]
@arledesma  I see, I misunderstood. You can do that via the CLI using `update-branch --environment-variables <value>` 

Ref: https://docs.aws.amazon.com/cli/latest/reference/amplify/update-branch.html

**Comment by hloriana** [comment]
@arledesma If the above suggestion didn't work for you, please feel free to re-open this issue. Thank you!

**Comment by arledesma** [comment]
@hloriana That did work. 

```shell
aws amplify update-branch --app-id $AWS_APP_ID --branch-name pr-41 --environment-variables ONLY_ON_PR_41=true
```

The command configured only the override.  The naming seems counterintuitive but is only a UI quirk.  👍 

![image](https://user-images.githubusercontent.com/4999353/143138916-12380c86-97b3-47f4-924a-62353a3df8b9.png)

Sorry that I did not respond earlier - I noticed the label `pending-investigation` and that there was no longer a `pending-customer-response` label so I made a bad assumption as to the status of the issue.

---

Is there any way to actually accomplish this in the Amplify Console UI? I am completely okay with performing this action via cli but the UI makes it appear as though it is not a supported feature. If there is not a method to accomplish this via the UI then I can open a feature request if required.

![image](https://user-images.githubusercontent.com/4999353/143138506-b464ca6e-b34e-432e-8d77-fce97ee1ebce.png)

If the default was not a disabled `All branches`, but instead a select like the override provides then the feature of adding only to a single branch would be aligned with the functionality provided by the api (using aws cli in this case).
